### PR TITLE
Support for reading alignments in specific region without CRAI index file

### PR DESCRIPTION
**Pull request by athos**
This PR allows the CRAM reader to read alignments from a specified region within a CRAM file without requiring an index file.

CRAM readers can instantly check if each container or slice overlaps with the specified region by examining their headers. This makes it much more efficient to read alignments by skipping non-overlapping containers or slices, rather than performing a linear scan of the entire CRAM file.

This PR also adds a new test file `medium_without_index.cram`, which is identical to `medium.cram` except that it does not come with its corresponding index file.

### CODE REVIEW

File: test/cljam/io/cram_test.clj [hunk 1]
```diff
@@ -39,11 +39,16 @@
...
-                                        {:reference common/hg19-twobit-file})]
+                                        {:reference common/hg19-twobit-file})
+                  cram-rdr' (cram/reader common/medium-without-index-cram-file
+                                         {:reference common/hg19-twobit-file})]
         (is (cram/indexed? cram-rdr))
-        (are [?region ?count] (= ?count (count (cram/read-alignments cram-rdr ?region)))
+        (is (not (cram/indexed? cram-rdr')))
+        (are [?region ?count] (= ?count
+                                 (count (cram/read-alignments cram-rdr ?region))
+                                 (count (cram/read-alignments cram-rdr' ?region)))
```
Comment (alumi) [inline test/cljam/io/cram_test.clj:51]: This test fails in my environment.
Maybe we need to reset the position of the mapped buffer before calling next `read-alignments`

Comment (athos) [reply test/cljam/io/cram_test.clj:51]: Thank you for pointing that out! Yeah, I'm just looking into it right now. I wasn't aware that I didn't run these tests on my local environment at all 😂

Comment (athos) [reply test/cljam/io/cram_test.clj:56]: b1b0ea4 should fix the position issue.
I also added 3623b37 to improve test coverage for the CRAM region read.

**Review by alumi — APPROVED** [review]
Thank you for fixing so quickly! LGTM 👍

**Review by niyarin — APPROVED** [review]
LGTM!

**Pull request merged**